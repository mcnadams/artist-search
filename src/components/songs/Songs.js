import React from 'react';
import PropTypes from 'prop-types';
import Song from './Song';

function Songs({ songs, artist }) {
  const songList = songs.map((song, i) => {
    return (
      <li key={i}>
        <Song song={song} artist={artist} />
      </li>
    );
  });

  return (
    <ul>
      {songList}
    </ul>
  );
}

Songs.propTypes = {
  songs: PropTypes.array.isRequired,
  artist: PropTypes.string.isRequired
};

export default Songs;
