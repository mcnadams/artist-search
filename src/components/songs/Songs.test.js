import React from 'react';
import { shallow } from 'enzyme';
import Songs from './Songs';

describe('Songs renders', () => {
  it('renders Songs', () => {
    const wrapper = shallow(<Songs artist={'lady'} songs={[{ date: 'yes', id: 'no' }]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
