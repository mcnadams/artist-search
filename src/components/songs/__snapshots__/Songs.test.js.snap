// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Songs renders renders Songs 1`] = `
<ul>
  <li
    key="0"
  >
    <Song
      artist="lady"
      song={
        Object {
          "date": "yes",
          "id": "no",
        }
      }
    />
  </li>
</ul>
`;
